src/main/scala/dpla/ingestion3/harvesters/api/IaHarvester.scala

```diff
@@ -0,0 +1,121 @@
+package dpla.ingestion3.harvesters.api
+
+import java.net.URL
+
+import dpla.ingestion3.confs.i3Conf
+import dpla.ingestion3.utils.HttpUtils
+import org.apache.http.client.utils.URIBuilder
+import org.apache.log4j.Logger
+import org.apache.spark.sql.SparkSession
+import org.json4s.DefaultFormats
+import org.json4s.jackson.JsonMethods.{compact, parse, render}
+
+import scala.util.{Failure, Success}
+
+class IaHarvester (spark: SparkSession,
+                   shortName: String,
+                   conf: i3Conf,
+                   outputDir: String,
+                   harvestLogger: Logger)
+  extends ApiHarvester(spark, shortName, conf, outputDir, harvestLogger) {
+
+  def mimeType: String = "application_json"
+
+  override protected val queryParams: Map[String, String] = Map(
+    "q" -> conf.harvest.query
+  ).collect{ case (key, Some(value)) => key -> value } // remove None values
+
+  override def localHarvest: Unit = {
+    implicit val formats: DefaultFormats.type = DefaultFormats
+
+    val iaCollections = conf.harvest.setlist.getOrElse("").split(",")
+
+    iaCollections.foreach(collection => {
+      // Mutable vars for controlling harvest loop
+      var continueHarvest = true
+      var cursor = ""
+
+      while(continueHarvest) getSinglePage(cursor, collection) match {
+        // Handle errors
+        case error: ApiError with ApiResponse =>
+          harvestLogger.error("Error returned by request %s\n%s\n%s".format(
+            error.errorSource.url.getOrElse("Undefined url"),
+            error.errorSource.queryParams,
+            error.message
+          ))
+          continueHarvest = false
+        // Handle a successful response
+        case src: ApiSource with ApiResponse =>
+          src.text match {
+            case Some(docs) =>
+              val json = parse(docs)
```

Comment:
Do you want to handle the possibility here that `parse(docs)` will throw an error if IA's JSON is malformed?  Or will it just bubble up in some useful way?  May not be worth worrying about if it's a low risk.  For that matter, is there a chance of an exception being thrown if the `identifier` and `cursor` aren't present where you expect them in the next few lines?

Reply:
I added some additional error handling in the latest commit